Firestore/core/src/firebase/firestore/nanopb/writer.cc

```diff
@@ -72,20 +37,77 @@ void Writer::WriteNanopbMessage(const pb_field_t fields[],
   }
 }
 
-ByteStringWriter::ByteStringWriter() : Writer({}) {
-  stream_ = WrapContainer(&buffer_);
+static bool AppendToBytesArray(pb_ostream_t* stream,
```

Comment:
Nit: prefer anonymous namespace (the `constexpr` constant above could be moved into it as well).